src/core/algorithms/dd/split/split.cpp

```diff
@@ -334,17 +386,16 @@ bool Split::IsFeasible(DF const& d) {
     return false;
 }
 
-std::vector<DF> Split::SearchSpace(model::ColumnIndex index) {
-    std::vector<DF> dfs;
-    DF d = min_max_dif_;
-    dfs.push_back(d);
+std::vector<DFConstraint> Split::IndexSearchSpace(model::ColumnIndex index) {
+    std::vector<DFConstraint> dfs;
+    dfs.push_back(min_max_dif_[index]);
 
     if (!has_dif_table_) {
         // differential functions should be put in this exact order for further reducing
         for (int i = num_dfs_per_column_ - 1; i >= 0; i--) {
-            if (i >= min_max_dif_[index].lower_bound && i < min_max_dif_[index].upper_bound) {
-                d[index] = {min_max_dif_[index].lower_bound, (double)i};
-                dfs.push_back(d);
+            if (model::GreaterOrEqual(i, min_max_dif_[index].lower_bound) &&
+                model::Less(i, min_max_dif_[index].upper_bound)) {
```

Comment:
I don't like the idea of performing those manual checks.
we can also extract this to a method of a DFConstraint to encapsulate the logic of respecting machine epsilon:
```cpp
bool IsWithinExclusive(double value) const {
        return GreaterOrEqual(value, lower_bound) && Less(value, upper_bound);
    }
```